### PR TITLE
[NTDLL:LDR] ldrapi.c: Minor code tweaks

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Clarify code.

Follow-up to bd0a549 (0.4.15-dev-6948).
See #5793.

## Proposed changes

- Minor code tweaks
- Add a FIXME comment.

### CODE REVIEW

File: dll/ntdll/ldr/ldrapi.c [hunk 1]
```diff
@@ -1000,6 +1000,8 @@ LdrQueryProcessModuleInformationEx(
     ANSI_STRING AnsiString;
     PCHAR p;
 
+    UNREFERENCED_PARAMETER(ProcessId);
```
Comment (learn-more) [inline dll/ntdll/ldr/ldrapi.c:1003]: This is an unimplemented code path, so it makes no sense to mark this as `UNREFERENCED_PARAMETER`

File: dll/ntdll/ldr/ldrapi.c [hunk 2]
```diff
@@ -1008,15 +1010,14 @@ LdrQueryProcessModuleInformationEx(
     _SEH2_TRY
     {
         /* Check if we were given enough space */
-        if (Size < UsedSize)
+        if (UsedSize > Size)
```
Comment (learn-more) [inline dll/ntdll/ldr/ldrapi.c:1013]: There is no point in this change.

File: dll/ntdll/ldr/ldrapi.c [hunk 3]
```diff
@@ -1083,8 +1084,7 @@ LdrQueryProcessModuleInformationEx(
                 ModulePtr++;
 
                 /* Increase number of modules */
-                if (ModuleInformation)
-                    ModuleInformation->NumberOfModules++;
+                ModuleInformation->NumberOfModules++;
```
Comment (learn-more) [inline dll/ntdll/ldr/ldrapi.c:1087]: This change is plain wrong.

File: dll/ntdll/ldr/ldrapi.c [hunk 4]
```diff
@@ -1098,6 +1098,7 @@ LdrQueryProcessModuleInformationEx(
     _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
     {
         /* Ignoring the exception */
+        // FIXME: It looks like STATUS_SUCCESS could be returned, though *ReturnedSize was not set, for example.
```
Comment (learn-more) [inline dll/ntdll/ldr/ldrapi.c:1101]: I personally don't see the point of you adding FIXME's in code that you do not seem to understand fully.

File: dll/ntdll/ldr/ldrapi.c [hunk 5]
```diff
@@ -1175,12 +1176,13 @@ LdrEnumerateLoadedModules(
...
     /* Release loader lock */
     Status = LdrUnlockLoaderLock(0, Cookie);
-    ASSERT(NT_SUCCESS(Status));
 
     /* Reset any successful status to STATUS_SUCCESS,
      * but leave failure to the caller */
     if (NT_SUCCESS(Status))
         Status = STATUS_SUCCESS;
+    else
+        ASSERT(NT_SUCCESS(Status));
```
Comment (learn-more) [inline dll/ntdll/ldr/ldrapi.c:1185]: This changes behavior in debug builds, and this change does not seem like a good change to me.

**Review by learn-more — CHANGES_REQUESTED** [review]
I don't see the point of this PR, and some changes are even plain wrong.

Unless some ldr/ntdll maintainer sees something worth keeping in this PR, I'd say we close it.

File: dll/ntdll/ldr/ldrapi.c (hunk 1 above)
Comment (RatinCN) [reply dll/ntdll/ldr/ldrapi.c:1003]: Print "Unimplement" instead seems better

File: dll/ntdll/ldr/ldrapi.c (hunk 3 above)
Comment (RatinCN) [reply dll/ntdll/ldr/ldrapi.c:1087]: See above:
`ModulePtr->LoadOrderIndex = ModuleInformation->NumberOfModules;`
If `ModuleInformation` is `NULL`, an exception will occurred and never reach here, so I think `if (ModuleInformation)` is superfluous and can be removed.

**Review by tkreuzer — CHANGES_REQUESTED** [review]
I agree with Mark, these changes are at best pointless.

**Pull request closed without merging**